File: 0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int Area = 0;
        int maximum_total_area = 0;
        int left = 0, right = height.size()-1;
        while (left < right){
            Area = (min(height[left], height[right]))*(right - left);
            maximum_total_area = max(Area, maximum_total_area);
            if(height[left]<height[right]){
                left ++;
            }else{
                right --;
            }
        }
        return maximum_total_area;
    }
};

File: 0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int l = 0, h = n-1;
        int maxArea = -1, area = 0;
        while(l<h){

            if(height[l]>height[h]){
                area = height[h]*(h-l);
                h--;
            }else{
                area = height[l]*(h-l);
                l++;
            }
           maxArea = Math.max(maxArea, area);
        }
        return maxArea;
    }
}
